s3-source-connector/README.md

```diff
@@ -21,7 +21,290 @@ published into the corresponding Kafka topic.
+- `aws_s3_region` - Name of the region for the bucket used for storing the records. Defaults to `us-east-1`.
+- `aws_s3_bucket` - Name of an existing bucket for storing the records. Mandatory.
+- `aws_s3_prefix` - The prefix that will be added to the file name in the bucket. Can be used for putting output files into a subdirectory.
+
+List of new configuration parameters:
+- `aws.access.key.id` - AWS Access Key ID for accessing S3 bucket.
+- `aws.secret.access.key` - AWS S3 Secret Access Key.
+- `aws.s3.bucket.name` - - Name of an existing bucket for storing the records. Mandatory. See bucket name rules: <https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html>
+- `aws.s3.endpoint` - The endpoint configuration (service endpoint & signing region) to be used for requests.
+- `aws.s3.prefix` - The prefix that will be added to the file name in the bucket. Can be used for putting output files into a subdirectory.
+- `aws.s3.region` - Name of the region for the bucket used for storing the records. Defaults to `us-east-1`.
+- `aws.sts.role.arn` - AWS role ARN, for cross-account access role instead of `aws.access.key.id` and `aws.secret.access.key`
+- `aws.sts.role.external.id` - AWS ExternalId for cross-account access role
+- `aws.sts.role.session.name` - AWS session name for cross-account access role
+- `aws.sts.role.session.duration` - Session duration for cross-account access role in Seconds. Minimum value - 900.
+- `aws.sts.config.endpoint` - AWS STS endpoint for cross-account access role.
+
+## Configuration
+
+[Here](https://kafka.apache.org/documentation/#connect_running) you can
+read about the Connect workers configuration and
+[here](https://kafka.apache.org/documentation/#connect_resuming), about
+the connector Configuration.
+
+Here is an example connector configuration with descriptions:
+
+```properties
+### Standard connector configuration
+
+## Fill in your values in these:
+
+## These must have exactly these values:
+
+# The Java class for the connector
+connector.class=io.aiven.kafka.connect.s3.AivenKafkaConnectS3SourceConnector
+
+# Number of worker tasks to run concurrently
+tasks.max=1
+
+# The key converter for this connector
+key.converter=org.apache.kafka.connect.storage.StringConverter
+
+# The type of data format used to write data to the kafka events.
+# The supported values are: `jsonl`, 'avro', `parquet` and 'bytes'.
+input.type=jsonl
+
+# A comma-separated list of topics to use as output for this connector
+# Also a regular expression version `topics.regex` is supported.
+# See https://kafka.apache.org/documentation/#connect_configuring
+topics=topic1,topic2
+
+# A comma-separated list of topic partitions where the connector's offset storage reader
+# can read the stored offsets for those partitions. If not mentioned, s3 objects will be read again if
+# available in the bucket
+topic.partitions=1,2,3
```

Comment:
This seem incorrect for the source connector. We should just read all the objects (ideally just once) is this a sink only config?

Reply:
This is for source connector, and based on these partitions, the offset storage reader reads the map